src/clp_ffi_js/ir/utils.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CLP_FFI_JS_IR_UTILS_HPP
+#define CLP_FFI_JS_IR_UTILS_HPP
+
+#include <algorithm>
+#include <cstddef>
+#include <type_traits>
+#include <vector>
+
+#include <clp/type_utils.hpp>
+#include <emscripten/val.h>
+
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/StreamReader.hpp>
+
+namespace clp_ffi_js::ir {
+
+/**
+ * Generates a filtered collection from all log events.
+ *
+ * @param log_level_filter Array of selected log levels.
+ * @param filtered_log_event_map A reference to `FilteredLogEventsMap` that stores filtered result.
+ * @param log_events
+ */
+template <typename LogEvents>
+auto filter_deserialized_events(
+        FilteredLogEventsMap& filtered_log_event_map,
+        LogLevelFilterTsType const& log_level_filter,
+        LogEvents const& log_events
+) -> void {
```

Comment:
Since the results will be stored in `filtered_log_event_map`, shall we move it to the end of the argument list and state that's an output parameter in the docstring?